App Crashed When I try to pick a document from storage in Android 10 or higher.

I tried to pick a pdf document from storage in android 10 or higher versions
below android 10 versions perfectly working

manged 
    <uses-permission android:name="android.permission.READ_MEDIA_IMAGES" />
    <uses-permission android:name="android.permission.READ_MEDIA_VIDEO" />
    <uses-permission android:name="android.permission.READ_MEDIA_AUDIO" />
permissions 
and I get. 

FATAL EXCEPTION: main
                                                                                                    Process: com.inaipi.spark, PID: 23383
                                                                                                    java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=907489456, result=-1, data=Intent { dat=content://com.android.providers.downloads.documents/... flg=0x43 }} to activity {com.inaipi.spark/com.nareshchocha.filepickerlibrary.ui.activitys.DocumentFilePickerActivity}: java.lang.NumberFormatException: For input string: "msf:1000000043"
                                                                                                    	at android.app.ActivityThread.deliverResults(ActivityThread.java:5323)
                                                                                                    	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5362)
                                                                                                    	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67)
                                                                                                    	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:201)
                                                                                                    	at android.os.Looper.loop(Looper.java:288)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7872)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
                                                                                                    Caused by: java.lang.NumberFormatException: For input string: "msf:1000000043"
                                                                                                    	at java.lang.Long.parseLong(Long.java:736)
                                                                                                    	at java.lang.Long.valueOf(Long.java:1188)
                                                                                                    	at com.nareshchocha.filepickerlibrary.utilities.FileUtils.getDownloadsDocumentPath(FileUtils.kt:114)
                                                                                                    	at com.nareshchocha.filepickerlibrary.utilities.FileUtils.pathFromURI(FileUtils.kt:35)
                                                                                                    	at com.nareshchocha.filepickerlibrary.utilities.FileUtils.getRealPath(FileUtils.kt:21)
                                                                                                    	at com.nareshchocha.filepickerlibrary.ui.activitys.DocumentFilePickerActivity$selectFile$1.invoke(DocumentFilePickerActivity.kt:66)
                                                                                                    	at com.nareshchocha.filepickerlibrary.ui.activitys.DocumentFilePickerActivity$selectFile$1.invoke(DocumentFilePickerActivity.kt:56)
                                                                                                    	at com.nareshchocha.filepickerlibrary.picker.PickerUtils.selectFile$lambda$0(PickerUtils.kt:23)
                                                                                                    	at com.nareshchocha.filepickerlibrary.picker.PickerUtils.$r8$lambda$5EjpwZuwUTQfOHixKEdyGZdAUG8(Unknown Source:0)
                                                                                                    	at com.nareshchocha.filepickerlibrary.picker.PickerUtils$$ExternalSyntheticLambda0.onActivityResult(Unknown Source:2)
                                                                                                    	at androidx.activity.result.ActivityResultRegistry.doDispatch(ActivityResultRegistry.java:418)
                                                                                                    	at androidx.activity.result.ActivityResultRegistry.dispatchResult(ActivityResultRegistry.java:375)
                                                                                                    	at androidx.activity.ComponentActivity.onActivityResult(ComponentActivity.java:793)
                                                                                                    	at androidx.fragment.app.FragmentActivity.onActivityResult(FragmentActivity.java:164)
                                                                                                    	at android.app.Activity.dispatchActivityResult(Activity.java:8628)
                                                                                                    	at android.app.ActivityThread.deliverResults(ActivityThread.java:5316)
                                                                                                    	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5362) 
                                                                                                    	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67) 
                                                                                                    	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:201) 
                                                                                                    	at android.os.Looper.loop(Looper.java:288) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7872) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936) 

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -1,9 +1,9 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 plugins {
-    id("com.android.application").version("8.0.2").apply(false)
-    id("com.android.library").version("8.0.2").apply(false)
-    kotlin("android").version("1.8.22").apply(false)
-    id("io.gitlab.arturbosch.detekt").version("1.23.0").apply(false)
+    id("com.android.application").version("8.1.1").apply(false)
+    id("com.android.library").version("8.1.1").apply(false)
+    kotlin("android").version("1.9.0").apply(false)
+    id("io.gitlab.arturbosch.detekt").version("1.23.1").apply(false)
 }
 
 tasks.register("clean", Delete::class) {

--- a/filepickerlibrary/build.gradle.kts
+++ b/filepickerlibrary/build.gradle.kts
@@ -3,13 +3,14 @@ plugins {
     id("com.android.library")
     id("org.jetbrains.kotlin.android")
     id("kotlin-parcelize")
+    id("kotlin-kapt")
     id("io.gitlab.arturbosch.detekt")
     id("maven-publish")
 }
 
 android {
     namespace = "com.nareshchocha.filepickerlibrary"
-    compileSdk = 33
+    compileSdk = 34
 
     defaultConfig {
         minSdk = 21
@@ -123,7 +124,7 @@ publishing {
 }
 
 detekt {
-    toolVersion = "1.23.0"
-    config = files("config/detekt/detekt.yml")
+    toolVersion = "1.23.1"
+    config.setFrom("$projectDir/config/detekt/detekt.yml")
     buildUponDefaultConfig = true
 }

--- a/filepickerlibrary/src/main/java/com/nareshchocha/filepickerlibrary/ui/activitys/DocumentFilePickerActivity.kt
+++ b/filepickerlibrary/src/main/java/com/nareshchocha/filepickerlibrary/ui/activitys/DocumentFilePickerActivity.kt
@@ -213,7 +213,7 @@ internal class DocumentFilePickerActivity : AppCompatActivity() {
 
         private fun getPermission(): String {
             return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
-                Manifest.permission.READ_MEDIA_IMAGES
+                Manifest.permission.READ_MEDIA_VIDEO
             } else {
                 Manifest.permission.READ_EXTERNAL_STORAGE
             }

--- a/filepickerlibrary/src/main/java/com/nareshchocha/filepickerlibrary/utilities/FileUtils.kt
+++ b/filepickerlibrary/src/main/java/com/nareshchocha/filepickerlibrary/utilities/FileUtils.kt
@@ -100,7 +100,7 @@ internal object FileUtils {
     private fun Context.getDownloadsDocumentPath(uri: Uri): String? {
         val fileName = getFilePath(this, uri)
         if (fileName != null) {
-            Environment.getExternalStorageDirectory()
+            return Environment.getExternalStorageDirectory()
                 .toString() + "/Download/" + fileName
         }
         var id = DocumentsContract.getDocumentId(uri)

--- a/sample/build.gradle.kts
+++ b/sample/build.gradle.kts
@@ -5,14 +5,14 @@ plugins {
 
 android {
     namespace = "com.nareshchocha.filepicker"
-    compileSdk = 33
+    compileSdk = 34
 
     defaultConfig {
         applicationId = "com.nareshchocha.filepicker"
         minSdk = 21
-        targetSdk = 33
-        versionCode = 1
-        versionName = "1.0"
+        targetSdk = 34
+        versionCode = 2
+        versionName = "1.1"
 
         testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
         testInstrumentationRunnerArguments.putAll(

--- a/sample/src/main/java/com/nareshchocha/filepicker/MainActivity.kt
+++ b/sample/src/main/java/com/nareshchocha/filepicker/MainActivity.kt
@@ -83,6 +83,7 @@ class MainActivity : AppCompatActivity() {
                     .pickDocumentFileBuild(
                         DocumentFilePickerConfig(
                             allowMultiple = true,
+
                             mMimeTypes = listOf("application/pdf", "image/*"),
                         ),
                     ),

